src/harness/sas_test_harness.py

```diff
@@ -284,71 +284,74 @@ def generateCbsdReferenceId(fcc_id, serial_number):
   """
   return str(fcc_id + '/' + str(hashlib.sha1(serial_number).hexdigest())).encode('utf-8')
 
-def generateCbsdRecords(cbsd_records, grant_records_list):
-  """ Generates the cbsdData Object by combining cbsdrecord and grantrecord based on SAS-SAS TS.
-
+def generateCbsdRecords(registration_requests, grant_requests_list):
+  """ Generates the cbsdData Object by combining registration request and grant request based
+      on SAS-SAS TS.
   Args:
-     cbsd_records: list of cbsdData objects that includes one or more grants object.
-     grant_records_list: list of muliple grant records for all cbsd devices.
+     registration_requests: list of registration requests that includes one or more grant requests.
+     grant_requests_list: list of muliple grant requests for all cbsd devices.
        For an example : muliple grants [ [grant1, grant2], [grant3, grant4 ] are
        associated with each cbsd records [ cbsd1, cbsd2 ].
 
   Returns:
-     It returns list of CBSDData Object based on SAS-SAS TS.
+     It returns list of CbsdData object based on SAS-SAS TS.
   """
 
   cbsd_records_list = []
-  for index, (cbsd_record, grant_records) in enumerate(zip(cbsd_records, grant_records_list)):
-    encoded_cbsd_id = generateCbsdReferenceId(cbsd_record['fccId'],
-                                         cbsd_record['cbsdSerialNumber'])
+  if not len(registration_requests) == len(grant_requests_list):
+    raise Exception('ConfigurationError: Number of registration requests doesnot match '
+                    'with number of grant requests')
+
```

Comment:
Consider also verifying that each list of grant requests (per CBSD) has at least one element?

Reply:
Extra verification added to ensure atleast one grant per CBSD is configured